### PR TITLE
Do not use Cython 3

### PR DESCRIPTION
**Pull request by mattwthompson**
https://github.com/pydantic/pydantic/blob/v1.10.12/requirements.txt#L3C38-L3C38

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mattwthompson** [comment]
Sure, will do

**Comment by mattwthompson** [comment]
These deps are not currently in the recipe; would you prefer they be added, added to `run_constrained`, or just left as-is? I'm inclined to leave out `dataclasses` since `python <3.7` is ancient, but I don't know anything about the other packages

```
devtools==0.9.0
email-validator==2.0.0.post2
dataclasses==0.6; python_version < '3.7'
python-dotenv==0.20.0

**Comment by swails** [comment]
Is cython really a run-time dependency?

**Comment by mattwthompson** [comment]
I think that's a matter of opinion https://github.com/pydantic/pydantic/blob/v1.10.12/pydantic/version.py

**Comment by swails** [comment]
I guess so.  Thanks!

I think this will resolve our problems with the newest pydantic release.

**Review by swails — APPROVED** [review]

**Comment by hmaarrfk** [comment]
hmm, thanks for taking look at this. i feel like the cython stuff can be patched out, but I didn't study their source in depth.

**Comment by hmaarrfk** [comment]
i generally don't think we should pin cython at runtime if we can avoid it for cond-forge.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,22 +12,22 @@ source:
...
     - {{ compiler('c') }}
   host:
     - python
     - pip
-    - cython
+    - cython =0.29.32
   run:
     - python
-    - typing-extensions >=4.2.0
+    - typing-extensions ==4.3.0
```
Comment (peterbygrave) [inline recipe/meta.yaml:30]: I think this should be kept at at >=4.2.0 to follow setup.py for run dependencies https://github.com/pydantic/pydantic/blob/d9c2af3a701ca982945a590de1a1da98b3fb4003/setup.py#L133

Comment (mattwthompson) [reply recipe/meta.yaml:30]: I'm just trying to follow what @pavelzw asked

**Comment by mattwthompson** [comment]
I can take 3d9757500ee0feb0e93a10371e5dfb8cc3b2d948 or leave it, and I can move Cython out of a runtime dependency.

What I can't do is merge this, and only the Cython change is needed to actually fix the most recent builds and the infrastructure that depends on it (numerous, to say the least). So that's really all I care about here

**Comment by hmaarrfk** [comment]
I would prefer it if we tried:
https://github.com/conda-forge/pydantic-feedstock/pull/98

**Comment by NoAnyLove** [comment]
Can we mark the current `1.10.12` build as broken until the fix is deployed?

**Comment by mattwthompson** [comment]
https://github.com/conda-forge/admin-requests#mark-packages-as-broken-on-conda-forge

**Comment by hmaarrfk** [comment]
Yes. The current version of pydantic 1..10.12 should be considered broken. I can press merge if the PR is made

**Comment by hmaarrfk** [comment]
I tested #98 and it addresses the symptoms I hit.

I'm weary about the need to add a test that would have caught this. we know that cython 3 isn't supported upstream.

this issue has been hitting the python ecosystem as a whole....

**Comment by mattwthompson** [comment]
Adding a test is just a suggestion for preventing the same issue from happening again. There are surely plenty of ways to safeguard against repeated broken builds, tests tend to be pretty lightweight

**Comment by hmaarrfk** [comment]
please see: https://github.com/conda-forge/pydantic-feedstock/pull/98 i think it now addresses the points above

**Comment by pavelzw** [comment]
Superseded by #98

**Pull request closed without merging**